### PR TITLE
Automated Installer & Directory Wrapper 

### PR DESCRIPTION
**Pull request by error-try-again**
Summary

- Code cleanup, refactoring & organization 
- Automated the procedure so you only need to run a one liner and it installs the plugin
- Integrated a deployment stack for generating a user specific Taildrop service that performs conflict resolution for file sending to the local machine from a remote
- Improved conditional checks, error checking & ping redirection
- Provided a directory wrapper solution so that directories can now be copied over in addition to files as individual archive's

### CODE REVIEW

**Comment by Stalloevan** [comment]
Thanks for the pull. I was looking to implement an automated installer, just haven't had time to get around to it yet. I'm going to review this and get back to you once I'm done.

**Comment by error-try-again** [comment]
> Thanks for the pull. I was looking to implement an automated installer, just haven't had time to get around to it yet. I'm going to review this and get back to you once I'm done.

No worries. You were most of the way there. It took about an hour and a half to implement and test this morning - if I’ve missed anything just let me know and I’ll make a patch.

File: install.sh [hunk 1]
```diff
@@ -81,70 +81,78 @@ generate_taildrop_script() {
...
-  if tailscale ping --timeout=0.01s "${chosen_device}" &>/dev/null; then
-    local list_names=""
-    # Process each file provided as argument
-    local file
-    for file in "$@"; do
-      if [[ -f "${file}" ]]; then # Ensure it's a file
-        tailscale file cp "${file}" "${chosen_device}":
-        # Append file name to list for notification
+  # Check if the chosen device is online
+  if ! tailscale ping --timeout=1s "${chosen_device}" &>/dev/null; then
```
Comment (Stalloevan) [inline install.sh:117]: The only change I would make is to change this to change this command from `if ! tailscale ping --timeout=1s "${chosen_device}" &>/dev/null; then` to `if ! tailscale ping --c 1 "${chosen_device}" &>/dev/null; then`. When I was working on the code originally, I couldn't use the c  flag to send 1 ping, but it seems to be working now. I only used the timeout flag as a workaround to maintain that 1 ping limit.

**Review by Stalloevan — CHANGES_REQUESTED** [review]
I left a few comments for review. Other than those, it should be all set to merge.

**Comment by error-try-again** [comment]
Awesome - I've updated the ping count limit `tailscale ping -c 1 "${chosen_device}"` and modified the link to direct to the kde website

**Pull request merged**